### PR TITLE
script: drop custom scripts

### PR DESCRIPTION
**Pull request by gdams**
fixes: https://github.com/nodejs/citgm/issues/301

### CODE REVIEW

**Comment by gibfahn** [comment]
Do we still need:
 - [lib/npm/test.js#L36-L37](https://github.com/gdams/citgm/blob/9a4b430ec726a584a4d0cb542d184530d8ef86ae/lib/npm/test.js#L36-L37)
```js
     if (data.scripts === undefined ||
         data.scripts.test === undefined) {
```
 - [README.md#L158](https://github.com/gdams/citgm/blame/9a4b430ec726a584a4d0cb542d184530d8ef86ae/README.md#L158) `and custom scripts.`

 - [lib/citgm.js#L31](https://github.com/gdams/citgm/blob/9a4b430ec726a584a4d0cb542d184530d8ef86ae/lib/citgm.js#L31) and L73.

```
31: * 4. Filter the output of 4 through the Reporter script (details TBD)
73: *  - options.reporter = path to a reporting script to run
```


The answer could be yes to all, I don't know.

**Comment by gdams** [comment]
@gibfahn we still need lib/npm/test.js#L36-L37, I have cleared up the readme

**Comment by gdams** [comment]
@MylesBorins I think I'm right in saying that lib/citgm.js#L31 and L73 is to do with the reporter rather than custom scripts?

File: test/bin/test-citgm-all.js [hunk 1]
```diff
@@ -17,17 +17,6 @@ test('citgm-all: /w markdown /w -j', function (t) {
   });
 });
 
-test('citgm-all: envVar /w -J', function (t) {
```
Comment (MylesBorins) [inline test/bin/test-citgm-all.js]: what does this have to do with script?

File: test/bin/test-citgm.js [hunk 2]
```diff
@@ -30,30 +30,6 @@ test('bin: omg-i-fail /w markdown output /w nodedir', function (t) {
   });
 });
 
-test('bin: omg-i-fail /w custom script /w tap to file /w junit to file /w append', function (t) {
```
Comment (MylesBorins) [inline test/bin/test-citgm.js]: make sure that we are not losing coverage on some of the options here. We may need to re-add those flags to other tests

File: test/bin/test-citgm-all.js (hunk 1 above)
Comment (gdams) [inline test/bin/test-citgm-all.js]: this env var test spawns a shell script and checks if an environment variable has been set

File: test/bin/test-citgm.js (hunk 2 above)
Comment (gdams) [inline test/bin/test-citgm.js]: okay so I have moved the extra options to other tests

**Comment by gdams** [comment]
@gibfahn @MylesBorins updated and rebased PTAL

**Comment by gdams** [comment]
@nodejs/citgm where are we with this? have we come to an agreement yet?

**Comment by gibfahn** [comment]
I'm +1 on this, could you rebase and make sure you've answered all @MylesBorins nits?

**Comment by gdams** [comment]
@gibfahn updated PTAL

**Comment by gibfahn** [comment]
You should change the title of the PR/commit though, this doesn't deprecate scripts, it removes them entirely.

**Comment by gdams** [comment]
CI is green, @targos can I grab a review from you please
